File: src/main/java/com/coleji/stockwatcher/controller/TestController.java
package com.coleji.stockwatcher.controller;

import com.coleji.stockwatcher.config.prop.AppProps;
import com.coleji.stockwatcher.model.polygon.DtoDividendResponse;
import com.coleji.stockwatcher.model.polygon.DtoOhlcResponse;
import com.coleji.stockwatcher.model.polygon.DtoSplitResponse;
import com.coleji.stockwatcher.util.JacksonUtil;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.jooq.DSLContext;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.*;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Stream;

@RestController("/")
class TestController {
	@Autowired
	private DSLContext dslContext;

	@Autowired
	private AppProps appProps;

	private final Logger logger = LoggerFactory.getLogger(this.getClass());

//	@GetMapping("/")
//	public String ping() {
//		Result<Record> r = dslContext.resultQuery("select * from s_p_daily_ohlc where ohlc_id = 1").fetch();
//
//		return r.getFirst().get("volume").toString();
//	}

	@GetMapping("/")
	public String ping() {
		List<File> allFiles = Stream.of(new File("out/sample/splits").listFiles())
			.filter(file -> !file.isDirectory())
			.sorted(Comparator.comparing(File::getName))
			.toList();

		int total = allFiles.size();
		int ct = 0;
		for (File f : allFiles) {
			try (FileInputStream fis = new FileInputStream(f)) {
				byte[] contents = fis.readAllBytes();
				ObjectMapper mapper = JacksonUtil.getObjectMapperBuilderWithModules()
						.build();
				DtoSplitResponse dto = mapper.readValue(contents, DtoSplitResponse.class);
				logger.info((++ct) + "/" + total + " - " + f.getName() + " - " + dto.getResults().size());
				if (ct==1) logger.info(dto.toString());
				break;
			} catch (IOException e) {
				throw new RuntimeException(e);
			}
		}

		return "pong";
	}
}

File: src/main/kotlin/com/coleji/stockwatcher/controller/TestController.kt
package com.coleji.stockwatcher.controller

import com.coleji.stockwatcher.config.AppProps
import com.coleji.stockwatcher.model.dto.DtoPolygonSplit
import com.coleji.stockwatcher.model.polygon.DtoSplitResponse
import com.coleji.stockwatcher.repository.PolygonSplitRepository
import com.coleji.stockwatcher.service.FidelityIngestService
import com.coleji.stockwatcher.util.JacksonUtil
import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletRequest
import org.jooq.DSLContext
import org.jooq.generated.tables.records.SPSplitsRecord
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import java.io.File
import java.io.FileInputStream
import java.io.IOException
import java.util.stream.Stream


@RestController("/")
class TestController(
	private val dslContext: DSLContext,
	private val appProps: AppProps,
	private val fidelityIngestService: FidelityIngestService,
	private val splitRepository: PolygonSplitRepository
) {
	private val logger: Logger = LoggerFactory.getLogger(this.javaClass)

	@GetMapping("/")
	fun ping(): String {
		return "pong"
	}

	@GetMapping("/splits-db")
	fun getAllSplits(): List<DtoPolygonSplit> {
		val results = splitRepository.getAll()
		return results.map { r -> DtoPolygonSplit.fromJooq(r) }
	}

	@PostMapping(value=["/import/fidelity-activity"], consumes = ["text/csv", "text/csv;charset=UTF-8"])
	fun fidelityImport(req: HttpServletRequest, @RequestBody bytes: ByteArray) {
		print(req.toString())
		fidelityIngestService.processImport(bytes)
	}

	@GetMapping("/splits")
	fun splits(): String {

		val allFiles = Stream.of<File>(*File("out/sample/splits").listFiles())
			.filter { file: File? -> !file!!.isDirectory() }
			.sorted(Comparator.comparing { obj: File? -> obj!!.getName() })
			.toList()

		val total = allFiles.size
		var ct = 0
		for (f in allFiles) {
			try {
				FileInputStream(f).use { fis ->
					val contents = fis.readAllBytes()
					print(String(contents))
					val mapper: ObjectMapper = JacksonUtil.getObjectMapperWithModules()
					val dto = mapper.readValue<DtoSplitResponse>(contents, DtoSplitResponse::class.java)
					logger.info((++ct).toString() + "/" + total + " - " + f.getName() + " - " + dto.results.size)
					if (ct == 1) logger.info(dto.toString())
				}
			} catch (e: IOException) {
				throw RuntimeException(e)
			}
		}

		return "pong"
	}
}